clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3865,13 +3865,13 @@ LangAS CodeGenModule::GetGlobalVarAddressSpace(const VarDecl *D) {
   }
 
   if (LangOpts.SYCLIsDevice) {
+    if (!D)
+      return LangAS::opencl_global;
```

Comment:
I understand that this is a useful bugfix but it seems to be a bit unrelated to this patch. Can we do it separately?

Reply:
That makes sense, I will do that.